Improving Commit-Reveal and PBS for MEV Resistance.md

```diff
@@ -0,0 +1,82 @@
+  
+ However, PBS does not prevent malicious MEV from users' transactions. To protect the transactions of users from MEV, privacy-preserving techniques like cryptography and zero-knowledge proof can make the details of transactions fully private except for users themselves, so builders and proposers have no chance to know the transaction content to launch MEV attacks on them. 
+  
+ However, the additional cost and delay caused by many privacy-persevering MEV resistance techniques are too high to be used in reality. Achieving full privacy-preserving like the zero-knowledge proof is too costly and inefficient, and there is no need to make transactions fully private for MEV resistance. Also, full privacy-preserving will inevitably compromise usability and transparency to some degree, which is not so applicable in practice like meeting requirements of the financial regulation. Full privacy also doesn't work in many DeFi applications like AMM, which needs to reveal the latest public states timely for usability and transparency. Other proposed solutions either need centralized third-party or strong security assumptions, which are not reliable and realistic under many situations. 
+  
+
+## Plan and Deliverables
+ To address the issues above, in our research, on the one hand, we will look into different kinds of privacy-preserving techniques like commit-reveal and optimize them to enhance their usability, efficiency, and reliability. On the other hand, by improving PBS and combining it with optimized privacy-preserving techniques, we make PBS MEV resistible, and improved PBS can better implement optimized privacy-preserving techniques. 
+  
+ In our commit-reveal design, users send encrypted transactions to proposers. Based on the improved PBS, proposers will first accept and reach consensus on them. They can't do any MEV attacks because the content of transactions is hidden. After proposers confirm which transactions will be put on chain, those transactions can be revealed and sent to the builder for validation and execution. Knowing the details of transactions after being revealed, builders can process them at lower cost and better efficiency, but they can't extract MEV because their order and content have already been determined by proposers during the commit phase. In this way, users could be protected from MEV attacks when proposers commit their transactions first for consensus, and maintain transparency and usability when builders reveal their transactions to process and put them on chain.
```

Comment:
Do you mean you are not sure if block consensus and building can be seperated done by different people? It have been designed and implemented by PBS. In PBS builders collect transactions, extract MEV and construct the block. They send their blocks to proposers and prospers will reach consensus and put them on chain. Actually PBS also uses commit-reveal to hide builders' transactions first, and only reveal when proposers accept them to prevent proposers from knowing and copying the MEV strategies builders make in advance. 

 The difference is that in our scheme, proposers reach consensus first and then builders construct the block accordingly. Commit-reveal is used to prevent proposers or builders from knowing the content of transactions beforehand and extracting MEV from them. Any other questions, feel free to ask me. Thanks a lot!

Reply:
How much latency does this introduce?

Reply:
Hi @sambacha Thanks for your kindly inquiry. 

We are still in research and design phase so unable to give accurate number now. We can get that after developing and testing a demo for our proposed solution given the grant. But theoretically we think no extra latency would be introduced. Compared with the current mining process, miners won't spend time on extracting MEV from users' transactions. Compared with current PBS scheme, builders and proposers don't need to spend time on committing and revealing back and forth. Compared with current commit-reveal scheme, users don't need to transactions on chain twice one for commit and one for reveal. Therefore at least we won't have more latency than other solutions.

Any other questions or critiques, feel free to ask me and we can improve our scheme accordingly. Thanks a lot!

Reply:
Knowing the transaction contents is important for knowing if the transaction has an expiry that may be invalid when the proposed block is executed for sealing, how do you get around this?

Reply:
When the builder executes the proposed block for sealing, he has already revealed and known the content. As the content and order of transactions have been determined by proposers first, the builder can't manipulate them for MEV.

Reply:
> When the builder executes the proposed block for sealing, he has already revealed and known the content. As the content and order of transactions have been determined by proposers first, the builder can't manipulate them for MEV.

a deadline on a transaction may be valid when the proposer includes it, but may not be valid any longer when the builder executes the block. I am not asking about manipulating MEV.

Reply:
Hi @ sambacha thanks for your kindly critiques, and sorry for no response during the weekend.

If you mean valid until block number, in our scheme proposers will accept the first block builder constructing the block and sending it to them. For instance, if a user appoints his transaction is valid only in new block #10, the first builder constructing the block #10 will be put on chain by proposers. 

Any other questions, feel free to ask me and we are designing the details of our scheme. Thanks a lot!

Reply:
hi @mryings thanks for your engagement so far on this!

> Do you mean you are not sure if block consensus and building can be seperated done by different people?
No, apologies if my comment was vague. I thought you were referring to the nodes that will do the execution as the builders - I realise now that you are actually proposing some significant changes to how the builder market would work.

To make sure I understand correctly, the proposed flow is as follows:
1. Users submit encrypted transactions to the proposer
2. Proposer builds a block locally
3. Committee comes to consensus on the block - doing very basic validation on transaction format that can be done within the limits of the encryption
4. Once the block is agreed upon, the proposer communicates with the block builder (this is the first communication between these two parties?)
5. The proposer tells the block builder which transactions it wants, and the builder builds a block from these transactions, therefore "revealing" them.

Is this along the lines of what you're planning?

Reply:
Hi @jldunne Thanks for your clarification. 

In our current scheme, proposer don't build the block. Proposers only make consensus on which transactions should be included in which order within the next block. It's like [crList in PBS proposal](https://notes.ethereum.org/@vbuterin/pbs_censorship_resistance). 

So there is no step2. Other steps are correct. After proposers reach consensus on the content and order of the new block, builders can reveal the committed transactions and construct the block according to proposers' consensus. Only builders construct the block following proposers' consensus can their block be accepted by proposers.

We will further redesign, revise, and validate our design afterwards. Anyway our final goal is to:
1. Protect users' interests and provide better trading experience and order for them
2. Build a more regulated and reliable on-chain process equal and fair to everyone. No one should have privileges including miners
3. Users and applications should have rights and freedom to self-determine the content and order of their transactions.

To achieve this goal we will also explore other feasible solutions to come up with the most suitable one. Thanks a lot!

Reply:
Dear Jolene Dunne:

Also we think the evaluation criteria of our solution could be:

1. The extent and degree of MEV attacks on users’ transactions we can avoid

2. The efficiency and cost of our malicious MEV prevention scheme

3. The level and scope of customization on transactions made by users and applications

4. The effectiveness of censorship resistance on users’ transactions

5. The amount of benefits our scheme can bring to users, applications and miners 

We also wonder about how long can we get the final result of our grant application? The details and plan of our research is highly negotiable and flexible, which can be discussed further and determined together after our application is approved. 

Thanks a lot!

Kindly Regards,
Jieliang Yin